realtime/utils.js

```diff
@@ -119,84 +120,63 @@ function isPresent(filterValueSet, objValueArr) {
 }
 
 /**
- * The filterString is used to extract the filterType and filter values and
- * the object is compared against the extracted filter to check if the field
- * of the object matches the filter criteria.
+ * The filterString is used to extract the filterType and filter values and the
+ * object is compared against the extracted filter to check if the field of the
+ * object matches the filter criteria.
+ *
  * @param  {String} filterString - String of the form filterType=values.
- * @param  {String|Array} objValues - The values of the object, that is to be
- * matched against a filter criteria
- * @returns {Boolean} - true if the object matches the filter criteria, false
- * otherwise.
+ * @param  {String|Array} objValues - The values of the object to be matched
+ *  against filter criteria (empty array if none)
+ * @returns {Boolean} - true if the object matches the filter criteria
  */
-function applyFilter(filterString, objValues) {
-  const objValueArr = [];
-  if (objValues && Array.isArray(objValues)) {
-    objValues.forEach((obj) => {
-      objValueArr.push(obj);
-    });
-  } else {
-    objValueArr.push(objValues);
-  }
+function applyFilter(filterString, objValues = []) {
+  // Short-circuit return true if there is no filterString
+  if (!filterString) return true;
 
   /*
-   * The filter string is of the form filterType=values. For example,
-   * an aspect filterString will be of the form INCLUDE=Temperature,Humidity
-   * where temperature and humidity are the aspects and INCLUDE is the
-   * filterType
+   * The filter string is a name-value pair of the form `filterType=values`.
+   * For example, aspect filterString `INCLUDE=Temperature,Humidity` has
+   * filterType `INCLUDE` and aspect values `Temperature` and `Humidity`.
+   * The "value" part of the name-value pair will be empty if the filter is not
+   * set in the perspective.
    */
-  if (filterString) {
-    const filterComponents =
-      filterString.split(constants.fieldTypeFieldSeparator);
+  const nvp = filterString.split(constants.fieldTypeFieldSeparator);
 
-    /*
-     * When the filters are not set the size of the filterComponents array is
-     * less than 2 and we are returning true,
-     *
-     */
-    if (filterComponents.length < 2) {
-      return true;
-    }
+  /*
+   * Short-circuit return true if the filterString is not a name-value pair
+   * with an "=" separator.
+   */
+  if (nvp.length < 2) return true;
 
-    // filter type is either INCLUDE or EXCLUDE
-    const filterType = filterComponents[0];
+  const filterType = nvp[0]; // INCLUDE or EXCLUDE
```

Comment:
I'd remove the comment since the line 153 explains the return when there is no match with INCLUDE/EXCLUDE.